PRChecker/Shared/GraphQL/NetworkService.swift

```diff
@@ -77,28 +61,34 @@ final class NetworkSerivce {
         let requestChainTransport = RequestChainNetworkTransport(interceptorProvider: provider, endpointURL: url)
 
         return ApolloClient(networkTransport: requestChainTransport, store: store)
-    }()
-    
-    func configure(for username: String, accessToken: String, endpoint: String, useLegacyQuery: Bool) {
-        self.username = username
-        self.accessToken = accessToken
-        self.apiEndpoint = endpoint
-        self.useLegacyQuery = useLegacyQuery
     }
     
     func getAllPRs() -> AnyPublisher<NetworkPRResult, Error> {
-        getAllPRs(for: username)
+        getAllPRs(for: SettingsViewModel.shared.loginViewModel.username)
     }
     
     func getAllPRs(for username: String) -> AnyPublisher<NetworkPRResult, Error> {
-        let assignedQuery = "is:pr assignee:\(username) archived:false sort:updated"
-        let requestedQuery = "is:pr review-requested:\(username) archived:false sort:updated"
-        let reviewedQuery = "is:pr reviewed-by:\(username) archived:false sort:updated"
         
-        return Publishers.Zip3(getPR(with: assignedQuery), getPR(with: requestedQuery), getPR(with: reviewedQuery))
-            .map { prLists in
-                (prLists.0 + prLists.1 + prLists.2)
-                    .arrayByRemovingDuplicates()
+        var publishers = [AnyPublisher<[AbstractPullRequest], Error>]()
+        
+        if SettingsViewModel.shared.displayOptions.contains(.assigned) {
+            let assignedQuery = "is:pr assignee:\(username) archived:false sort:updated"
+            publishers.append(getPR(with: assignedQuery))
+        }
+        if SettingsViewModel.shared.displayOptions.contains(.reviewRequested) {
+            let requestedQuery = "is:pr review-requested:\(username) archived:false sort:updated"
+            publishers.append(getPR(with: requestedQuery))
+        }
+        if SettingsViewModel.shared.displayOptions.contains(.reviewed) {
+            let reviewedQuery = "is:pr reviewed-by:\(username) archived:false sort:updated"
+            publishers.append( getPR(with: reviewedQuery))
+        }
+        
+        guard !publishers.isEmpty else { return Fail(error: NetworkServiceError.missingQuery).eraseToAnyPublisher() }
+        
+        return Publishers.MergeMany(publishers)
```

Comment:
Changing to use `Merge` causes a small UX issue where if more than 1 option is checked, when refreshing the list, all PRs will disappear for a while before new result loading, I think this screen flicker should be improved if possible.

Reply:
Hmm... let me take a look.

Reply:
Got it. We needed to `collect()` the results since it was expecting a complete collection (from Zip) but was getting it piecemeal from MergeMany